src/main/kotlin/de/smartsquare/squit/mediatype/xml/XmlCanonicalizer.kt

```diff
@@ -29,14 +29,38 @@ class XmlCanonicalizer : Canonicalizer {
         return if (mediaTypeConfig.xmlCanonicalize) {
             val outputStream = ByteArrayOutputStream()
 
-            canonicalizer.canonicalize(input.toByteArray(), outputStream, false)
+            val content = if (mediaTypeConfig.resolveInvalidNamespaces)
+                resolveInvalidNamespaces(input, mediaTypeConfig.resolveNamespaceString)
+            else
+                input
+
+            canonicalizer.canonicalize(content.toByteArray(), outputStream, false)
 
             SAXReader().read(outputStream.toByteArray().inputStream()).asString()
         } else {
             input
         }
     }
 
+    private fun resolveInvalidNamespaces(content: String, resolveNamespaceString: String): String {
+        var contentReplaced = content
+        val regexNamespaces = Regex("xmlns([^\\s!>]+)")
```

Comment:
Please move the regexes to `private val` in a new `private companion object` to avoid compiling them each time this function is run.